### PR TITLE
Fix nullable attribute of SlotDescriptor

### LINKED ISSUE #5262: query sql crash
### Steps to reproduce the behavior (Required)
```
DROP DATABASE IF EXISTS h5;
CREATE DATABASE h5;
USE h5;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 DATETIME NOT NULL ) DUPLICATE KEY (c_0_0) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 91 day)) DISTRIBUTED BY HASH (c_0_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 DECIMAL(2, 1)  NULL ,c_1_1 DATETIME NULL ,c_1_2 CHAR(21) NULL ,c_1_3 DATETIME MIN NULL ,c_1_4 VARCHAR(1) REPLACE_IF_NOT_NULL NULL ,c_1_5 VARCHAR(1) MAX NOT NULL ,c_1_6 DATE MIN NULL ,c_1_7 PERCENTILE PERCENTILE_UNION NOT NULL ,c_1_8 DECIMAL(29, 2)  SUM NULL ,c_1_9 VARCHAR(11) MAX NOT NULL ,c_1_10 PERCENTILE PERCENTILE_UNION NULL ,c_1_11 DOUBLE REPLACE NOT NULL ) AGGREGATE KEY (c_1_0,c_1_1,c_1_2) PARTITION BY RANGE(c_1_1) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 109 day)) DISTRIBUTED BY HASH (c_1_2,c_1_0) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t0 VALUES ("2010-01-28 08:00:14");
TRUNCATE TABLE t0;
INSERT INTO t0 VALUES ("2010-01-08 08:00:14");
INSERT INTO t1 VALUES (NULL, "2010-01-11 08:00:14", "E", NULL, "", "", "2010-01-07", percentile_hash("847506242"), 3270552573658715567184174.23, "yvN", percentile_hash("1439817050"), 0.497959738938638), (0.1, NULL, "aw", "2010-01-02 08:00:14", NULL, "", "2010-01-07", percentile_hash("847506242"), 4668494930696124689539772.13, "5", percentile_hash("847506242"), 0.22194113326156328), (0.8, "2010-01-29 08:00:14", "jwZyd", "2010-01-31 08:00:14", "", "", "2010-01-29", percentile_hash("-1634749188"), 5040007184431755660192209.23, "U", percentile_hash("-1634749188"), 0.6383141568517576);
INSERT INTO t1 VALUES (0.0, "2010-01-13 08:00:14", NULL, "2010-01-18 08:00:14", "", "", "2009-12-31", percentile_hash("-1238627201"), 8391848110740879440605510.80, "D", percentile_hash("1439817050"), 0.6383141568517576);
CREATE VIEW v0 AS SELECT t1.c_1_0, t1.c_1_1, t1.c_1_2, t1.c_1_3, t1.c_1_4, t1.c_1_5, t1.c_1_6, t1.c_1_8, t1.c_1_9, t1.c_1_11, t0.c_0_0 FROM t1, t0 WHERE (NOT (false)) GROUP BY t1.c_1_0, t1.c_1_1, t1.c_1_2, t1.c_1_3, t1.c_1_4, t1.c_1_5, t1.c_1_6, t1.c_1_8, t1.c_1_9, t1.c_1_11, t0.c_0_0 ORDER BY t1.c_1_3, t1.c_1_8, t1.c_1_5, t1.c_1_8, t1.c_1_11, t1.c_1_5;
DELETE FROM t0 WHERE c_0_0="1969-12-29 01:58:48";
SELECT subt0.c_0_0 FROM t1,  (SELECT t0.c_0_0 FROM t0) subt0 LEFT ANTI JOIN v0 ON subt0.c_0_0 = v0.c_1_5 AND subt0.c_0_0 < v0.c_1_5 AND subt0.c_0_0 >= v0.c_1_11 AND subt0.c_0_0 >= v0.c_1_11 GROUP BY subt0.c_0_0, subt0.c_0_0, subt0.c_0_0 HAVING MIN( ( subt0.c_0_0 ) <= ( (SELECT t0.c_0_0 FROM v0, t0, t1 WHERE (v0.c_1_2)  BETWEEN ((("l")||(""))) AND (v0.c_1_2) )  ) )  ORDER BY subt0.c_0_0 ASC;


```
```
crash.log:
*** Check failure stack trace: ***
    @          0xa00eee4  google::LogMessage::Fail()
    @          0xa00ee43  google::LogMessage::SendToLog()
    @          0xa00e80c  google::LogMessage::Flush()
    @          0xa011ad8  google::LogMessageFatal::~LogMessageFatal()
    @          0x8001737  starrocks::vectorized::MaxMinAggregateFunction<>::update()
    @          0x8002af7  starrocks::vectorized::AggregateFunctionBatchHelper<>::update_batch()
    @          0x7286ab6  starrocks::Aggregator::compute_batch_agg_states()
    @          0x70b9e27  starrocks::pipeline::AggregateStreamingSinkOperator::_push_chunk_by_auto()
    @          0x709c6c8  starrocks::pipeline::AggregateStreamingSinkOperator::push_chunk()
    @          0x704e051  starrocks::pipeline::PipelineDriver::process()
    @          0x70262f2  starrocks::pipeline::GlobalDriverExecutor::_worker_thread()
    @          0x70255cc  _ZZN9starrocks8pipeline20GlobalDriverExecutor10initializeEiENKUlvE_clEv
    @          0x702bb96  _ZSt13__invoke_implIvRZN9starrocks8pipeline20GlobalDriverExecutor10initializeEiEUlvE_JEET_St14__invoke_otherOT0_DpOT1_
    @          0x702b664  _ZSt10__invoke_rIvRZN9starrocks8pipeline20GlobalDriverExecutor10initializeEiEUlvE_JEENSt9enable_ifIX16is_invocable_r_vIT_T0_DpT1_EES6_E4typeEOS7_DpOS8_
    @          0x702b0be  _ZNSt17_Function_handlerIFvvEZN9starrocks8pipeline20GlobalDriverExecutor10initializeEiEUlvE_E9_M_invokeERKSt9_Any_data
    @          0x43a42c2  std::function<>::operator()()
    @          0x59c4c5c  starrocks::FunctionRunnable::run()
    @          0x59c1790  starrocks::ThreadPool::dispatch_thread()
    @          0x59ded84  std::__invoke_impl<>()
    @          0x59de667  std::__invoke<>()
    @          0x59dda7e  _ZNSt5_BindIFMN9starrocks10ThreadPoolEFvvEPS1_EE6__callIvJEJLm0EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE
    @          0x59dc5ea  std::_Bind<>::operator()<>()
    @          0x59d94d0  std::__invoke_impl<>()
    @          0x59d7498  _ZSt10__invoke_rIvRSt5_BindIFMN9starrocks10ThreadPoolEFvvEPS2_EEJEENSt9enable_ifIX16is_invocable_r_vIT_T0_DpT1_EESA_E4typeEOSB_DpOSC_
    @          0x59d2c33  std::_Function_handler<>::_M_invoke()
    @          0x43a42c2  std::function<>::operator()()
    @          0x59abd13  starrocks::Thread::supervise_thread()
    @     0x7ffb5c947ea5  start_thread
    @     0x7ffb5bf628dd  __clone
    @              (nil)  (unknown)


```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - branch-2.2 ASAN
 - mysql> select current_version();
+------------------------------+
| current_version()            |
+------------------------------+
| SQLANCER_01_ASAN e59d04a |
+------------------------------+

### PR DESCRIPTION
**Pull request by liuyehcf**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5262

## Problem Summary(Required) ：

```sql
SELECT c_0_0, count(*)
FROM t0
GROUP BY c_0_0
HAVING ANY_VALUE(c_0_0 <= (
	SELECT c_0_0
	FROM t0
	WHERE c_0_0 > '2030-01-01'
));
```

The following expression maybe nullable

```sql
c_0_0 <= (
	SELECT c_0_0
	FROM t0
	WHERE c_0_0 > '2030-01-01'
)
```

### CODE REVIEW

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/analysis/FunctionCallExpr.java|2|2|100.00%|

**Review by Seaven — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ReplaceColumnRefRewriter.java|2|2|100.00%|

**Comment by liuyehcf** [comment]
run starrocks_fe_sonar

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ReplaceColumnRefRewriter.java [hunk 1]
```diff
@@ -45,6 +45,10 @@ public ScalarOperator visitVariableReference(ColumnRefOperator column, Void cont
         // Rewiring predicate shouldn't change the origin project columnRefMap
 
         ScalarOperator mapperOperator = operatorMap.get(column).clone();
+        // Keep the original attribute of nullable
+        if (mapperOperator.isColumnRef()) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ReplaceColumnRefRewriter.java:49]: This should be the clone method wrong.  we should keep nullable info when clone.

Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ReplaceColumnRefRewriter.java:49]: Column is the key, clone performed on value. I think we cannot change some attributes(isNullable is one of them) when replacing columnRef with scalarOperator.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/SubqueryTransformer.java|2|2|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|15|15|100.00%|

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|15|15|100.00%|

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 12 / 13 (92.31%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|11|11|100.00%|
|:large_blue_circle:|com/starrocks/analysis/BinaryPredicate.java|1|1|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Comment by kangkaisen** [comment]
Please  update the PR tittle.

**Review by Seaven — APPROVED** [review]

**Comment by liuyehcf** [comment]
@Mergifyio backport branch-2.2

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 12 / 13 (92.31%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|11|11|100.00%|
|:large_blue_circle:|com/starrocks/analysis/BinaryPredicate.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 12 / 13 (92.31%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/Expr.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|11|11|100.00%|
|:large_blue_circle:|com/starrocks/analysis/BinaryPredicate.java|1|1|100.00%|

**Pull request merged**